src/FSharp.Stats/Algebra/LinearAlgebra.fs

```diff
@@ -219,6 +219,96 @@ module LinearAlgebra =
         //                    else LinearAlgebraManaged.QR a
         LinearAlgebraManaged.QR a
 
+    /// Performs QR decomposition using an alternative algorithm.
+    /// Returns the orthogonal matrix Q and the upper triangular matrix R.
+    let qrAlternative (A: Matrix<float>) =
+        let m: int = A.NumRows
+        let n: int = A.NumCols
+
+        let q: Matrix<float> = Matrix.zero m n
+        let r: Matrix<float> = Matrix.zero n n
+        let qLengths: Vector<float> = Vector.zeroCreate n
+
+        let getVectorLength (v: Vector<float>) = Vector.fold (fun folder i -> folder+(i*i)) 0. v
+
+        let setqOfA (n: int) =
+            let aN: Vector<float> =  Matrix.getCol A n
+            let qN = 
+                if n = 0 then 
+                    aN 
+                else 
+                    Array.init (n) (fun i -> 
+                        let denominator = qLengths[i]
+                        let forNominator: Vector<float> = Matrix.getCol q i 
+                        let nominator: float = Vector.dot aN forNominator
+                        r.[i, n] <- nominator
+                        (nominator/denominator) * forNominator
+                    )
+                    |> Array.fold (fun folder  e -> folder-e ) aN
+            Matrix.setCol q n qN
+            qN  
+
+        for i=0 to n-1 do
+            let qN = setqOfA i 
+            let qLength = getVectorLength qN
+            let rValue = sqrt(qLength)
+            r[i,i] <- rValue
+            qLengths[i] <- qLength
```

Comment:
I do not have a direct reference since I had a look at the mathematical explanation for QR via Gram Schmidt and tried implementing it myself. The only real upside using this method instead of the in FSharp.Stats established one is the difference in the output Dimensions in R. An actual reference would be the numpy implantation of the reduced qr https://numpy.org/doc/stable/reference/generated/numpy.linalg.qr.html .

I also think that this implemtation needs to be optimised in the future to perfect the GLM.

Reply:
> I'm not sure if I get whats happening here:
> 
> 1. qLengths is a zero-vector that gets filled iteratively for each `i` in line 262
> 2. in line 256 setqOfA is called with the current i. However in l 247 qLength is called at position `i` which always should be 0, because it was initialized with 0. Just after the determination of qLength the value at this specific index is replaced in line 262.
> 
> Obviously I miss a step, but is there a possibility to create the qLength by Vector.init (...) rather than keeping it mutable and accessing/mutating it at multiple positions? If not, I'm happy to merge it, but I got confused during my attempt to understand whats happening here..

The mutation was done to simplify the function to match the mathematical explanation better. This is one of the things that could be updated when I get to it.